### PR TITLE
Add proposal about the common libraries concept in Kyma

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add proposal about libraries in Kym

**Related issue(s)**

- https://github.com/kyma-project/community/issues/142

### CODE REVIEW

File: sig-and-wg/sig-core/proposals/common-libraries.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+In the future, we should execute the next phase and review our libraries. We should also check if we are ready to extract common libraries outside and announce them as the official SDK for all Kyma internal and external developers. 
+
+### Package name 
+
+In Go ecosystem, there is a pretty popular `pkg` layout pattern. Many well-known projects, such as [Kubernetes](https://github.com/kubernetes/kubernetes/tree/master/pkg), [Docker](https://github.com/moby/moby/tree/master/pkg), [Grafana](https://github.com/grafana/grafana/tree/master/pkg), [Influxdb](https://github.com/influxdata/influxdb/tree/master/pkg), [Etcd](https://github.com/etcd-io/etcd/tree/master/pkg), use this pattern. 
+However, the `pkg` directory is used mainly for public libraries. They can be used internally by your application but also by external projects. 
+This is an informal contract between you and the external users of your code. 
+
+In Go, there is an ability to hide code using internal directories. If you put your code in an `internal` directory, no external project will be able to import that code.
```
Comment (klaudiagrz) [inline sig-and-wg/sig-core/proposals/common-libraries.md]:
```suggestion
In Go, there is a possibility to hide code using internal directories. If you put your code in an `internal` directory, no external project will be able to import that code. 
```

File: sig-and-wg/sig-core/proposals/common-libraries.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+### Package name 
+
+In Go ecosystem, there is a pretty popular `pkg` layout pattern. Many well-known projects, such as [Kubernetes](https://github.com/kubernetes/kubernetes/tree/master/pkg), [Docker](https://github.com/moby/moby/tree/master/pkg), [Grafana](https://github.com/grafana/grafana/tree/master/pkg), [Influxdb](https://github.com/influxdata/influxdb/tree/master/pkg), [Etcd](https://github.com/etcd-io/etcd/tree/master/pkg), use this pattern. 
+However, the `pkg` directory is used mainly for public libraries. They can be used internally by your application but also by external projects. 
+This is an informal contract between you and the external users of your code. 
+
+In Go, there is an ability to hide code using internal directories. If you put your code in an `internal` directory, no external project will be able to import that code. 
+Initially, the idea was to put our private libraries in `kyma/internal/pkg/{private_libs}` but we consolidated all project to one repository, so there is no option to vendor such libraries in our project. In this case, when we execute the `dep ensure` command, we receive such error:
```
Comment (klaudiagrz) [inline sig-and-wg/sig-core/proposals/common-libraries.md]:
```suggestion
Initially, the idea was to put our private libraries in `kyma/internal/pkg/{private_libs}` but we consolidated all the project to one repository, so there is no option to vendor such libraries in our project. In this case, when we execute the `dep ensure` command, we receive such error:  
```

File: sig-and-wg/sig-core/proposals/common-libraries.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ 
+Another example is to unify the logging approach in our projects. This will be also resolved by the other proposal in which the source code will be placed in the common libraries package, so all projects can simply use it.  
+
+## Consequences
+
+Once approved, these are the consequences:
+- new package for common libraries must be created and maintained
+- new expert CODEOWNERS must be defined for reviewing libraries added for common package
+- new package must be announced and other projects should switch to use the official libraries instead of custom implementation
+- developers always need to check our libraries not to reimplement the business logic
```
Comment (klaudiagrz) [inline sig-and-wg/sig-core/proposals/common-libraries.md]:
```suggestion
- developers must always check our libraries not to reimplement the business logic
```

File: milv.config.yaml [hunk 4]
```diff
@@ -22,6 +22,10 @@ files:
     config:
       #milv doesn't support parenthesis in parenthesis in .md links yet
       white-list-external: ["https://github.com/kyma-project/community/blob/master/docs/cla/SAP%20Corporate%20Contributor%20License%20Agreement%20(5-26-15"]
+  - path: "./community/sig-and-wg/sig-core/proposals/common-libraries.md"
+    config:
+      #milv doesn't support anachors in links https://github.com/magicmatatjahu/milv/issues/9
```
Comment (pkosiec) [inline milv.config.yaml]: `anchors`

File: milv.config.yaml [hunk 5]
```diff
@@ -22,6 +22,10 @@ files:
     config:
       #milv doesn't support parenthesis in parenthesis in .md links yet
       white-list-external: ["https://github.com/kyma-project/community/blob/master/docs/cla/SAP%20Corporate%20Contributor%20License%20Agreement%20(5-26-15"]
+  - path: "./community/sig-and-wg/sig-core/proposals/common-libraries.md"
+    config:
+#milv doesn't support anchors in links https://github.com/magicmatatjahu/milv/issues/9
```
Comment (mszostok) [inline milv.config.yaml]:
```suggestion
	#milv doesn't support anchors in links https://github.com/magicmatatjahu/milv/issues/9
```

**Pull request merged**